python/pyarrow/tests/test_compute.py

```diff
@@ -1702,13 +1702,37 @@ def test_logical():
 
 
 def test_cast():
+    arr = pa.array([1, 2, 3, 4], type='int64')
+    options = pc.CastOptions(pa.int8())
+
+    with pytest.raises(ValueError):
+        pc.cast(arr, target_type=None)
+
+    with pytest.raises(ValueError):
+        pc.cast(arr, 'int32', safe=True, options=options)
+
+    with pytest.raises(ValueError):
+        pc.cast(arr, 'int32', safe=False, options=options)
+
+    assert pc.cast(arr, 'int8', options=options) == pa.array(
+        [1, 2, 3, 4], type='int8')
+
     arr = pa.array([2 ** 63 - 1], type='int64')
+    options = pc.CastOptions(pa.int32())
+    allow_overflow_options = pc.CastOptions(
+        pa.int32(), allow_int_overflow=True)
 
     with pytest.raises(pa.ArrowInvalid):
         pc.cast(arr, 'int32')
 
+    with pytest.raises(pa.ArrowInvalid):
+        pc.cast(arr, 'int32', options)
```

Comment:
Ah - this is misleading, this is passing `safe=options`, is that what you meant to test?

Reply:
This was intended to ensure that passing a `CastOptions` object like this would fail in the same way `safe=options` would.

Reply:
What I mean is that even though we're passing `options`, it's getting ignored/interpreted as `safe=True` (and fails due to a value being out of range). It looks like a weird thing to test, basically.